### ISSUE TITLE
Add form validation rule for min & max size range

### ISSUE THREAD

**Issue by VanquishedWombat** [fomantic/Fomantic-UI#1279]
Issue: Form validation does not include a range length rule. This is bad because I have to apply both the minLength __AND__ maxLength rules and if the user causes a validation failure then the message they see only informs them of the specific rule that they broke. For a field with an implied input size range, this is bad UX - we should inform the user of both limits of size in the same message.

Proposal: Add a size validation rule:

size            : '{name} must be between {ruleValue[0]} and {ruleValue[1]} characters in length',

Example - see screenName parameter below.

$('.ui.form')
  .form({
    fields: {
      name     : 'empty',
      gender   : 'empty',
      username : 'empty',
      password : ['minLength[6]', 'empty'],
      skills   : ['minCount[2]', 'empty'],
      terms    : 'checked',
      __screenName__    : __'size[10,20]'__,
    }
  })
;

Note: this requires ruleValue to handle an array of values and therefore may be more than a simple change.

**Comment by prudho** [comment]
Here's a quick implementation that you can include in your code to implements it:
```js
$.fn.form.settings.rules.size = function(value, range) {
  var
    intRegExp = $.fn.form.settings.regExp.integer,
    min,
    max,
    parts
  ;
  if( !range || ['', '..'].indexOf(range) !== -1) {
    // do nothing
  }
  else if(range.indexOf('..') == -1) {
    if(intRegExp.test(range)) {
      min = max = range - 0;
    }
  }
  else {
    parts = range.split('..', 2);
    if(intRegExp.test(parts[0])) {
      min = parts[0] - 0;
    }
    if(intRegExp.test(parts[1])) {
      max = parts[1] - 0;
    }
  }

  return (
    (min === undefined || value.length >= min) &&
    (max === undefined || value.length <= max)
  );
}
```

The `size` rule is now defined and can be implemented using the short way:
```js
screenName : 'size[10..20]'
```
... or the long way:
```js
screenName: {
  rules: [
    {
      type: 'size[10..20]',
      prompt: '{name} length should be between 10 and 20!'
    }
  ]
}
```

This not perfect tho, but I think this is a good start.

**Comment by lubber-de** [comment]
Implemented by #2688
